13 entity cart

lyubovkirpa opened the pull request.

lyubovkirpa committed:
HCF-13 Cart.

lyubovkirpa committed:
HCF-13 warnings.

AIT-Reutow commented on src/main/java/ait/cohort49/hostel_casa_flamingo/model/entity/Cart.java:
```diff
@@ -0,0 +1,122 @@
+        this.user = user;
+    }
+
+    @Override
```
перемести пожалуйста эти методы в низ класса

AIT-Reutow commented on src/main/java/ait/cohort49/hostel_casa_flamingo/service/CartServiceImpl.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    @Override
+    public void clearCart(Cart cart) {
```
лучше передавать сюда в аргументом мета cardId. И выполнить поиск корзины по id,
Либо дождаться когда будет готова авторизация и сделать поиск корзины по авторизованному пользователю

AIT-Reutow commented on src/main/java/ait/cohort49/hostel_casa_flamingo/service/interfaces/CartService.java:
```diff
@@ -0,0 +1,22 @@
+
+import java.math.BigDecimal;
+
+public interface CartService {
```
А что если нам пока вообще отказаться от сервиса и контроллера для сущности Cart. И реализовать когда будет готова security

AIT-Reutow committed:
Merge branch 'master' of https://github.com/lyubovkirpa/hostel_Casa_Flamingo into 13-Entity-Cart

AIT-Reutow committed:
Refactor CartController and related services for cleanup.

Standardized CartController endpoints by removing user ID path variables and simplifying method logic. Updated Cart and User entities to establish a bidirectional one-to-one relationship. Refactored CartService and its implementation to improve modularity, introduce CartRepository, and streamline operations on user carts.

AIT-Reutow committed:
Merge branch 'master' of https://github.com/lyubovkirpa/hostel_Casa_Flamingo into 13-Entity-Cart
